core/src/main/java/org/apache/carbondata/core/readcommitter/TableStatusReadCommittedScope.java

```diff
@@ -79,13 +83,24 @@ public TableStatusReadCommittedScope(AbsoluteTableIdentifier identifier,
   @Override
   public Map<String, String> getCommittedIndexFile(Segment segment) throws IOException {
     Map<String, String> indexFiles;
-    if (segment.getSegmentFileName() == null) {
+    SegmentFileStore fileStore =
```

Comment:
based on the old check if segmentFileName can be null then this code can fail..Please check this